Hifi 20069

MarcelEdward opened the pull request:
This are the updated files for 20069

I got into a merge error and somehow made thing worse trying to fix it.

hifi-gustavo commented:
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

cozza13 commented:
ok to test

murillodigital commented:
add to whiltelist

cozza13 commented:
ok to test

MarcelEdward commented:
ok to test

murillodigital commented:
ok to test

hifi-gustavo commented:
Build console output for commit: c83c51e179b69131e99ca40d40855cf83d594676 available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

hifi-gustavo commented:
Build console output for commit: 171bac7f31d7d4b9f24546d91c1d2bde073670cf available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

SamGondelman commented:
Windows:
    - Worked briefly, then stopped working, no matter how many times I restart Interface or check/uncheck the menu option (which should probably be called "Activate 3D Connexion Devices" or something more clear).  When it worked briefly, though, I couldn't yaw, pitch, or zoom.  Registering/unregistering the device seems to be fixed.

Mac:
    - I couldn't get it to work at all.  I have the driver installed and the menu option was checked.  Did you get it to successfully work on mac while testing?

Also you should close https://github.com/highfidelity/hifi/pull/5261, but make sure this PR has all the changes you made over there, and finish making the changes Andrew mentioned.  And it looks like you have some conflicts so you might have to merge from master again.  Let me know if you need any help with that to prevent issues!

SamGondelman commented on interface/src/devices/3Dconnexion.cpp:
```diff
@@ -0,0 +1,1026 @@
+        _axisStateMap[makeInput(ROTATION_AXIS_Y_POS).getChannel()] = (cc_rotation.y > 0.0f) ? cc_rotation.y / MAX_AXIS : 0.0f;
+        _axisStateMap[makeInput(ROTATION_AXIS_Y_NEG).getChannel()] = (cc_rotation.y < 0.0f) ? -cc_rotation.y / MAX_AXIS : 0.0f;
+    } else {
+        _axisStateMap[makeInput(POSITION_AXIS_X_POS).getChannel()] = (cc_position.x > 0.0f) ? cc_position.x / MAX_AXIS : 0.0f;
```
You should be setting the axes regardless of the buttons.  When you add the input channel itself, you can give it two inputs.  See next comment.

SamGondelman commented on interface/src/devices/3Dconnexion.cpp:
```diff
@@ -0,0 +1,1026 @@
+    // these only get filled when button 1 is pressed
+    mapper.addInputChannel(UserInputMapper::PITCH_DOWN, makeInput(ROTATION_AXIS_Y_NEG), JOYSTICK_PITCH_SPEED);
+    mapper.addInputChannel(UserInputMapper::PITCH_UP, makeInput(ROTATION_AXIS_Y_POS), JOYSTICK_PITCH_SPEED);
+
```
Here, you can do mapper.addInputChannel(UserInputMapper::YAW_RIGHT, makeInput(ROTATION_AXIS_Z_POS), makeInput(BUTTON_1), JOYSTICK_YAW_SPEED); to make pitch require BUTTON_1 without hardcoding it above.

MarcelEdward commented:
When I download the source from https://github.com/MarcelEdward/hifi/tree/hifi-20069 and compile it, then it does work. 
The dmg installer does not work.

Is the 3Dconnexion dmg from http://www.3dconnexion.eu/service/drivers.html installed on the jenkins build server ?
Or is jenkins gonna recognize it as a framework when I copy it in ?

hifi-gustavo commented:
Build console output for commit: 3d612f2605af34ac59604a959b21b01b80b02f58 available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

hifi-gustavo commented:
Build console output for commit: 1668e39315bcb20abee01a7edc6f52a2943d39ef available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

SamGondelman commented:
Windows:
    - Buttons still don't work, but everything else seems great!

Mac:
    - Everything works (if I compile from your branch directly).

I had forgotten that I needed to compile it directly from your branch on mac, sorry.  Regardless, it doesn't seem like your framework stuff is working, so I would recommend getting rid of it.  I'll work on getting the driver installed on our build server.

Consider switching the pitch controls to the other rotation axis so they make more sense.  Also, it's a little hard to rotate while pressing the button anyways, so maybe just make BUTTON_1 + forward/backward translation control pitch and BUTTON_1 + left/right translation control yaw.  Play around with it and find the mapping that is easiest to use.

Looks like you still need to merge from master and resolve any conflicts, so if you could do that and try to clean up your style (mostly making sure the indentation is still correct everywhere), that would be great!

hifi-gustavo commented:
Build console output for commit: f6e0bdad35d69563d28763655bc3f71372bf2598 available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

hifi-gustavo commented:
Build console output for commit: d1b6954c995fe73ee5c8cdcfbb1fec0906524bc3 available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

hifi-gustavo commented:
Build console output for commit: 696b8bb52623c506e59dd1311ca61be880c8ba8b available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

hifi-gustavo commented:
Build console output for commit: b5c0dcbe3bd3b0df37387ac4ca9a4dd613363859 available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

MarcelEdward commented:
Put the zooming again on button_1 and button_2
it is a hard to press both buttons at the same time, almost ever
button_1 of button_2 gets called
It is very hard to press a button and at the same time move or rotate
the device, so I removed the simultaneous button and rotation.

the head in now moved down or up with tilting forward or backward. It
is a bit hard to do this without moving the avatar a bit.

The buttons should now work on windows

reverted the adding of the framework since this does not work

I am not sure if I resolved the merge conflict correct. But hifi-gustavo says all checks have passed :)

SamGondelman commented:
On windows zooming is super fast and turning doesn't seem to work but on mac it still works great!

And it looks like you still have some merge conflicts, are you sure you're syncing your branch properly?  It looks like your recent merge might have messed something up, github thinks you changed 378 files.

hifi-gustavo commented:
Build console output for commit: a06860289a81febb35b6f18a068540e587d5899e available at: https://gist.github.com/7076bbbd997a760d61d2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5351.dmg

MarcelEdward commented:
rollback to the commit before I added the framework and the #4, looks like I did not sync the branch properly. Let's try to figure out the git commands to do this.

I should do  merge changes from origin master to the origin development branch, but those changes should not end up here.
